Project/Source/Utils/Trail.cpp

```diff
@@ -118,43 +113,52 @@ void Trail::OnEditorUpdate() {
 	if (ImGui::Checkbox("Render", &isRendering)) {
 		isStarted = false;
 	}
-	ImGui::DragFloat("Witdh", &width, App->editor->dragSpeed2f, 0, inf);
+	ImGui::DragFloat("Width", &width, App->editor->dragSpeed2f, 0, inf);
 	if (ImGui::DragInt("Trail Quads", &trailQuads, 1.0f, 1, 50, "%d", ImGuiSliderFlags_AlwaysClamp)) {
 		if (nTextures > trailQuads) nTextures = trailQuads;
 		DeleteQuads();
 	}
-	if (ImGui::DragInt("Texture Repeats", &nTextures, 1.0f, 1, trailQuads, "%d", ImGuiSliderFlags_AlwaysClamp)) {
-		DeleteQuads();
-		EditTextureCoords();
-	}
-
 	ImGui::DragFloat("Quad Life", &quadLife, App->editor->dragSpeed2f, 1, inf);
 
+	ImGui::NewLine();
 	ImGui::Checkbox("Color Over Trail", &colorOverTrail);
 	if (colorOverTrail) {
 		ImGui::GradientEditor(gradient, draggingGradient, selectedGradient);
 	}
 
+	ImGui::NewLine();
 	ImGui::ResourceSlot<ResourceTexture>("texture", &textureID);
-	ResourceTexture* textureResource = App->resources->GetResource<ResourceTexture>(textureID);
-	if (textureResource != nullptr) {
-		int width;
-		int height;
-		glGetTextureLevelParameteriv(textureResource->glTexture, 0, GL_TEXTURE_WIDTH, &width);
-		glGetTextureLevelParameteriv(textureResource->glTexture, 0, GL_TEXTURE_HEIGHT, &height);
-
-		ImGui::NewLine();
-		ImGui::Separator();
-		ImGui::TextColored(App->editor->titleColor, "Texture Preview");
-		ImGui::TextWrapped("Size:");
-		ImGui::SameLine();
-		ImGui::TextWrapped("%d x %d", width, height);
-		ImGui::Image((void*) textureResource->glTexture, ImVec2(200, 200));
-		ImGui::Separator();
+	ImGui::Text("Flip: ");
+	ImGui::SameLine();
+	ImGui::Checkbox("X", &flipTexture[0]);
+	ImGui::SameLine();
+	ImGui::Checkbox("Y", &flipTexture[1]);
+
+	if (ImGui::DragInt("Texture Repeats", &nTextures, 1.0f, 1, trailQuads, "%d", ImGuiSliderFlags_AlwaysClamp)) {
+		DeleteQuads();
+		EditTextureCoords();
 	}
+
+	ImGui::NewLine();
+	ImGui::Indent();
+	if (ImGui::CollapsingHeader("Texture Preview")) {
+		ResourceTexture* textureResource = App->resources->GetResource<ResourceTexture>(textureID);
+		if (textureResource != nullptr) {
+			int width;
+			int height;
+			glGetTextureLevelParameteriv(textureResource->glTexture, 0, GL_TEXTURE_WIDTH, &width);
+			glGetTextureLevelParameteriv(textureResource->glTexture, 0, GL_TEXTURE_HEIGHT, &height);
+
+			ImGui::TextWrapped("Size:");
+			ImGui::SameLine();
+			ImGui::TextWrapped("%d x %d", width, height);
+			ImGui::Image((void*) textureResource->glTexture, ImVec2(200, 200));
+		}
+	}
+	ImGui::Unindent();
 }
 
-void Trail::Draw() {
+void Trail::Draw(float4x4 modelMatrix) {
```

Comment:
This should probably be passed by const reference to avoid unnecessary copies.

Reply:
Actually, it's not necessary since it's not used for now. I will keep Draw() empty and I will put it again with the stretch implementation